### PR TITLE
Adding a win32 platform for new mach Core

### PR DESCRIPTION
**Pull request by hordurj**
Adding a win32 platform for new mach Core.
Changes to Sprites and Text module to work with d3d12.

- [X] By selecting this checkbox, I agree to license my contributions to this project under the license(s) described in the LICENSE file, and I have the right to do so or have received permission to do so by an employer or client I am producing work for whom has this right.

### CODE REVIEW

File: src/gfx/Sprite.zig [hunk 1]
```diff
@@ -87,8 +90,12 @@ fn updatePipeline(
             // that all entities with the same pipeline value are stored in contiguous memory, and
             // skip this copy.
             if (sprite_pipeline_id == pipeline_id) {
+                const uv = uv_transform;
                 gfx.SpritePipeline.cp_transforms[i] = transform;
-                gfx.SpritePipeline.cp_uv_transforms[i] = uv_transform;
+                gfx.SpritePipeline.cp_uv_transforms[i].v[0] = vec4(uv.v[0].x(), uv.v[0].y(), uv.v[0].z(), 0.0);
+                gfx.SpritePipeline.cp_uv_transforms[i].v[1] = vec4(uv.v[1].x(), uv.v[1].y(), uv.v[1].z(), 0.0);
+                gfx.SpritePipeline.cp_uv_transforms[i].v[2] = vec4(uv.v[2].x(), uv.v[2].y(), uv.v[2].z(), 0.0);
+                gfx.SpritePipeline.cp_uv_transforms[i].v[3] = vec4(0.0, 0.0, 0.0, 0.0);
```
Comment (slimsag) [inline src/gfx/Sprite.zig:98]: Can you keep the old line with a TODO comment? Something like:

```
// TODO(d3d12): we should be able to just write this but for some reason it doesn't
// work with d3d12 backend:
// gfx.SpritePipeline.cp_uv_transforms[i] = uv_transform;
...your code...
```

File: src/gfx/Sprite.zig [hunk 2]
```diff
@@ -99,7 +106,7 @@ fn updatePipeline(
     // Sort sprites back-to-front for draw order, alpha blending
     const Context = struct {
         transforms: []Mat4x4,
-        uv_transforms: []Mat3x3,
+        uv_transforms: []Mat4x4,
```
Comment (slimsag) [inline src/gfx/Sprite.zig:109]: Same thing here

**Review by alichraghi — APPROVED** [review]
Amazing!

File: src/gfx/TextPipeline.zig [hunk 3]
```diff
@@ -134,6 +134,7 @@ pub const Glyph = extern struct {
 
     /// Which text this glyph belongs to; this is the index for transforms[i], colors[i].
     text_index: u32,
+    text_padding: u32,
```
Comment (slimsag) [inline src/gfx/TextPipeline.zig:137]: same here

**Comment by ypsvlq** [comment]
For the parts of win32/Core.zig that were based on my work:

- [x] By selecting this checkbox, I agree to license my contributions to this project under the license(s) described in the LICENSE file, and I have the right to do so or have received permission to do so by an employer or client I am producing work for whom has this right.

**Pull request merged**